src/handlers/dashboard/get_chart_data.py

```diff
@@ -91,21 +98,27 @@ def _format_payload(df: pandas.DataFrame, query_params: dict, filters: list) ->
     return payload
 
 
-# @generic_error_handler(msg="Error retrieving chart data")
+@decorators.generic_error_handler(msg="Error retrieving chart data")
 def chart_data_handler(event, context):
     """manages event from dashboard api call and retrieves data"""
     del context
     query_params = event["queryStringParameters"]
     filters = event["multiValueQueryStringParameters"].get("filter", [])
     path_params = event["pathParameters"]
     boto3.setup_default_session(region_name="us-east-1")
-    query = _build_query(query_params, filters, path_params)
-    df = awswrangler.athena.read_sql_query(
-        query,
-        database=os.environ.get("GLUE_DB_NAME"),
-        s3_output=f"s3://{os.environ.get('BUCKET_NAME')}/awswrangler",
-        workgroup=os.environ.get("WORKGROUP_NAME"),
-    )
-    res = _format_payload(df, query_params, filters)
-    res = http_response(200, res)
+    try:
+        query = _build_query(query_params, filters, path_params)
+        df = awswrangler.athena.read_sql_query(
+            query,
+            database=os.environ.get("GLUE_DB_NAME"),
+            s3_output=f"s3://{os.environ.get('BUCKET_NAME')}/awswrangler",
+            workgroup=os.environ.get("WORKGROUP_NAME"),
+        )
+        res = _format_payload(df, query_params, filters)
+        res = functions.http_response(200, res)
+    except errors.AggregatorS3Error:
+        res = functions.http_response(
+            404, f"Aggregate for {path_params['data_package_id']} not found"
```

Comment:
nit: that's a very specific conclusion to draw from a generic "something happened error". Might be helpful to log the original error message and/or include it here too.

Reply:
The error should get logged to cloudwatch, security asked us to have no stack traces in responses and to generally have these error messages be unhelpful while the API is externally accessible.

Reply:
Maybe add a comment to that effect? Like... if a third developer comes across this in 2 years, they won't have that context. Something like "# don't be too informative about internal state in our error messages" (but in Matt voice) above this generic response.

Reply:
ok, added a brief note.